### PR TITLE
Test coverage improvements

### PR DESCRIPTION
**Pull request by atombrella**
PLEASE DON'T MERGE! There are still things that need to be worked on!

`get_context_data` does not need to call `get_form` in sub-classes of `FormView`. You can check this when you read the coverage report.

### CODE REVIEW

File: tests/core/test_views.py [hunk 1]
```diff
@@ -594,3 +597,60 @@ def test_merge_preview(self):
...
+    def test_content(self):
+        response = self.client.get(reverse('wiki:history', kwargs={
+            'article_id': self.root_article.pk,
+        }))
+        self.assertIn('History:', str(response.content))
+        self.assertEqual(response.context['selected_tab'], 'history')
+
+
+class SettingsViewTests(RequireRootArticleMixin, ArticleWebTestUtils, DjangoClientTestBase):
+    # TODO needs fixing
```
Comment (atombrella) [inline tests/core/test_views.py]: The test runner gives a 404 here.

File: tests/core/test_views.py [hunk 2]
```diff
@@ -343,6 +343,9 @@ def test_articles_cache_is_cleared_after_deleting(self):
         # test the cache
         self.assertContains(self.get_by_path('TestCache/'), 'Content 2')
 
+    # def test_dont_delete_children(self):
```
Comment (atombrella) [inline tests/core/test_views.py]: Test data needs to be set up here.

**Comment by atombrella** [comment]
https://stackoverflow.com/questions/58189683/typeerror-attrib-got-an-unexpected-keyword-argument-convert perhaps it's time to update the `pytest` version in `setup.py` (the file needs a few updates, e.g., to remove `Python 3.4`).

**Comment by benjaoming** [comment]
> `get_context_data` does not need to call `get_form` in sub-classes of `FormView`. You can check this when you read the coverage report.

@atombrella the more I look at old CBV code, the more I get a clear sense that multi-inheritence and mixins were a blessing in disguise :)

I even found complete breakage in the `DeleteView`, mainly caused by lack of testing: We tested that `POST` data to the page worked, but not that the form would actually render in the `GET` request... which it didn't :smile: 

I've added bit and pieces to this PR and it's really some awesome ground work you laid out to improve the test coverage. Thanks!

**Pull request merged**